deployment/huggingface/yolos-seq2seq.ipynb

```diff
@@ -0,0 +1,1699 @@
+            "53dace0191ae40018f47a5e14caf9a66",
+            "33495a661db74624be41bad46b526d90",
+            "a733a8c125dc46bcb23efc024ee60a6c"
+          ]
+        },
+        "id": "CPBadtBsiKpn",
+        "outputId": "047eb733-5239-4de7-e43e-e374a81abdff"
+      },
+      "outputs": [],
+      "source": [
+        "yolos = Yolos()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "R6jR843MiXtc"
+      },
+      "outputs": [],
+      "source": [
+        "url = yolos.example()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "6lMHliQ1iY2Y"
+      },
+      "outputs": [],
+      "source": [
+        "predictions = yolos.predict(url)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Z4gX4jxvkFA8"
+      },
+      "outputs": [],
+      "source": [
+        "results_rows = []\n",
+        "\n",
+        "for item in predictions[0]:\n",
+        "    item = item.copy()\n",
+        "    item.update(item.pop('box'))\n",
+        "    results_rows.append(pd.Series(item))"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Nb7dcFUsiZC4"
+      },
+      "outputs": [],
+      "source": [
+        "df = pd.concat(results_rows, axis=1).T"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 426
+        },
+        "id": "IVCKPhFMq25v",
+        "outputId": "e0cdf59d-1828-42b7-807c-6c63be974a57"
+      },
+      "outputs": [],
+      "source": [
+        "df"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "KgGilFLFfgiw"
+      },
+      "source": [
+        "## 6. Model Register"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "gqTwdp7lhVlS",
+        "outputId": "b8cc3f44-0356-4020-ce94-585add15e8ef"
+      },
+      "outputs": [],
+      "source": [
+        "registered_model = client.get_or_create_registered_model(name='YOLOS - Hugging Face')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "iDCNSz-zhjdj",
+        "outputId": "66babeb0-a96b-4a24-e245-690417a048e0"
+      },
+      "outputs": [],
+      "source": [
+        "model = registered_model.create_standard_model(\n",
+        "      name = 'v0',\n",
+        "      model_cls = Yolos,\n",
+        "      model_api = ModelAPI([{'urls': yolos.example()}], yolos.predict(yolos.example())[0]),\n",
```

Comment:
Could we use

```python
model_api = ModelAPI(yolos.example(), yolos.predict(yolos.example())),
```

to match the model's input/outpput format?

Reply:
Sure, but it changes how WebApp displays information (as we discussed yesterday).

The API `output` ends up with the `VertaList` containing a `VertaJson` for each object found, indexing them with numbers:
```
{
  "type": "VertaList",
  "name": "",
  "value": [
    {
      "type": "VertaJson",
      "name": "0",
      "value": [
        {
          "type": "VertaJson",
          "name": "box",
          "value": [
            {
              "type": "VertaFloat",
              "name": "xmax"
            },
            {
              "type": "VertaFloat",
              "name": "xmin"
            },
            {
              "type": "VertaFloat",
              "name": "ymax"
            },
            {
              "type": "VertaFloat",
              "name": "ymin"
            }
          ]
        },
        {
          "type": "VertaString",
          "name": "label"
        },
        {
          "type": "VertaFloat",
          "name": "score"
        }
      ]
    }
  ]
}
```

And the **Input & Output** section ends up like this:
![image](https://user-images.githubusercontent.com/2164940/210566253-46138d57-1228-4e4a-8d70-3b3d2fbdf082.png)

Is that ok?

Reply:
That is fine. As I've mentioned before, _the most important thing_ is that `ModelAPI` matches _exactly_ the model's input and output.

The WebApp might not look great, but it is better than having an incorrect API.

Reply:
If we would like to make things a _bit_ nicer in the UI, I would recommend these steps:

1. change `example()` to return
   ```python
   [{'url': 'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_1.jpg'}]
   ```
   so the WebApp will show `"name": "url"`
2. For `model_api=` use
   ```python
   ModelAPI(yolos.example(), [[yolos.predict(yolos.example())[0][0]]])
   ```
   which I don't personally like, but would make the API `"output"` in the WebApp much simpler.

Reply:
Hum,
 `ModelAPI(yolos.example(), [[yolos.predict(yolos.example())[0][0]]])`
is much more confusing than
`ModelAPI(yolos.example(), yolos.predict(yolos.example())`

I'll keep it as how it is now, specially since you tracked this front-end improvement in Jira :+1: